--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -238,7 +238,7 @@
 )
 
 install(
-    FILES "${project_config}" "${project_version}" "packaging/cmake/Modules/FindLZ4.cmake"
+    FILES "${project_config}" "${project_version}"
     DESTINATION "${config_install_dir}"
 )
 
--- a/packaging/cmake/Config.cmake.in
+++ b/packaging/cmake/Config.cmake.in
@@ -23,10 +23,6 @@
   endif()
 endif()
 
-if(@WITH_LZ4_EXT@)
-  find_dependency(LZ4)
-endif()
-
 find_dependency(Threads)
 
 include("${CMAKE_CURRENT_LIST_DIR}/@targets_export_name@.cmake")
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -191,7 +191,7 @@
 target_link_libraries(rdkafka PUBLIC Threads::Threads)
 
 if(WITH_SASL_CYRUS)
-  target_link_libraries(rdkafka PUBLIC ${SASL_LIBRARIES})
+  target_link_libraries(rdkafka PRIVATE ${SASL_LIBRARIES})
 endif()
 
 if(WITH_LIBDL)
@@ -199,7 +199,7 @@
 endif()
 
 if(WITH_LZ4_EXT)
-  target_link_libraries(rdkafka PUBLIC LZ4::LZ4)
+  target_link_libraries(rdkafka PRIVATE LZ4::LZ4)
 endif()
 
 # Set up path to these sources for other sub-projects (tests, examples)
