Index: librdkafka-0.11.4/src/rd.h
===================================================================
--- librdkafka-0.11.4.orig/src/rd.h
+++ librdkafka-0.11.4/src/rd.h
@@ -158,8 +158,8 @@ static RD_INLINE RD_UNUSED char *rd_strn
 #ifdef __APPLE__
 /* Some versions of MacOSX dont have IOV_MAX */
 #define IOV_MAX 1024
-#elif defined(_MSC_VER)
-/* There is no IOV_MAX on MSVC but it is used internally in librdkafka */
+#elif defined(_MSC_VER) || defined(__GNU__)
+/* There is no IOV_MAX on MSVC or GNU but it is used internally in librdkafka */
 #define IOV_MAX 1024
 #else
 #error "IOV_MAX not defined"
Index: librdkafka-0.11.4/src/snappy_compat.h
===================================================================
--- librdkafka-0.11.4.orig/src/snappy_compat.h
+++ librdkafka-0.11.4/src/snappy_compat.h
@@ -5,7 +5,7 @@
 
 #ifdef __FreeBSD__
 #  include <sys/endian.h>
-#elif defined(__APPLE_CC_) || defined(__MACH__)  /* MacOS/X support */
+#elif defined(__APPLE_CC_) || (defined(__MACH__) && defined(__APPLE__))  /* MacOS/X support */
 #  include <machine/endian.h>
 
 #if    __DARWIN_BYTE_ORDER == __DARWIN_LITTLE_ENDIAN
