Description: fix typos
 files don't exist on ppc64el
Author: Andreas Beckmann <anbe@debian.org>
Forwarded: not-needed

--- a/nvidia-cuda/include/cudaEGL.h
+++ b/nvidia-cuda/include/cudaEGL.h
@@ -257,7 +257,7 @@ typedef struct CUeglStreamConnection_st*
  * glFinish command on all GLcontexts (for OpenGL and likewise for other APIs).
  * The application will be also responsible for ensuring that any pending operation on the
  * registered CUDA resource has completed prior to executing subsequent commands in other APIs
- * accesing the same memory objects.
+ * accessing the same memory objects.
  * This can be accomplished by calling cuCtxSynchronize or cuEventSynchronize (preferably).
  *
  * The surface's intended usage is specified using \p flags, as follows:
@@ -346,7 +346,7 @@ CUresult CUDAAPI cuEGLStreamConsumerConn
  * The EGLStreamKHR is an EGL object that transfers a sequence of image frames from one
  * API to another.
 
- * \param conn            - Conection to disconnect.
+ * \param conn            - Connection to disconnect.
  *
  * \return
  * ::CUDA_SUCCESS,
@@ -439,7 +439,7 @@ CUresult CUDAAPI cuEGLStreamProducerConn
  * The EGLStreamKHR is an EGL object that transfers a sequence of image frames from one
  * API to another.
 
- * \param conn            - Conection to disconnect.
+ * \param conn            - Connection to disconnect.
  *
  * \return
  * ::CUDA_SUCCESS,
