Author: Gianfranco Costamagna <locutusofborg@debian.org>
Last-Update: 2020-02-03

--- gst-plugins-bad1.0-1.16.2.orig/configure.ac
+++ gst-plugins-bad1.0-1.16.2/configure.ac
@@ -1845,7 +1845,7 @@ AG_GST_CHECK_FEATURE(OPENCV, [opencv plu
       HAVE_OPENCV="yes"
     fi
   ], [
-    PKG_CHECK_MODULES([OPENCV], [opencv4 >= 4.0.0 opencv4 < 4.2.0] , [
+    PKG_CHECK_MODULES([OPENCV], [opencv4 >= 4.0.0 opencv4 < 4.3.0] , [
     AC_PROG_CXX
     AC_LANG([C++])
     OLD_CPPFLAGS=$CPPFLAGS
